pom.xml

```diff
@@ -2166,7 +2165,7 @@
                     <artifactId>modernizer-maven-plugin</artifactId>
                     <configuration>
                         <violationsFiles>
-                            <violationsFile>${air.main.basedir}/.mvn/modernizer/violations.xml</violationsFile>
+                            <violationsFile>${session.rootDirectory}/.mvn/modernizer/violations.xml</violationsFile>
```

Comment:
It does not work like this, you need an "indirection". This property is evaluated ONLY in Maven CLI params! (CLI or those coming from `maven.config`).

So indirection would be: in maven.config set
`-Dmodernizer.violationsFile=${session.rootDirectory}/.mvn/modernizer/violations.xml`

and in here use
`<violationsFile>modernizer.violationsFile</violationsFile>`

Reason is how Maven4 got many other related changes in this are, but the "backport" to Maven 3.9 was "mild" or "limited". So ONLY MAVEN CLI PARAMS are interpolated. But you can circumvent it with indirection like thiat and it will still work with Maven4 as well.

Reply:
Thanks, let me try that!